csv-data-management/extract_data_from_csv.py

```diff
@@ -470,34 +513,41 @@ def _df_to_csv(df, filename=None):
         id = str(uuid4().fields[-1])
         file_name = filename + ".csv" if filename else "dataset-{}.csv".format(id)
         compression = dict(method='zip', archive_name=file_name)
-        df.to_csv('/tmp/{}'.format(file_name.split(".")[0]) + '.zip', encoding='utf-8', header='true',
-                  compression=compression, index=False)
-        filepath = '/tmp/{}'.format(file_name.split(".")[0]) + '.zip'
+
+        if isinstance(df, pd.DataFrame):
+            df.to_csv('/tmp/{}'.format(file_name.split(".")[0]) + '.zip', encoding='utf-8', header='true',
+                      compression=compression, index=False)
+            filepath = '/tmp/{}'.format(file_name.split(".")[0]) + '.zip'
+
+        else:
```

Comment:
what exactly is df in this case?

Reply:
also, why no compression in this case?

Reply:
Here, Dataframe can be of pandas and polars (Pandas df only in case of Convert JSON to CSV, JSON Field action) , As there was no direct way of Compression in Polars write_csv action, I had to make two conditions if the DF is of Pandas or Polars.